packages/toolpad-app/src/toolpad/AppEditor/PagePanel.tsx

```diff
@@ -95,16 +50,33 @@ export default function PagePanel({ appId, className, sx }: ComponentPanelProps)
           alignItems: 'center',
         }}
       >
-        {isLoading ? (
+        {isLoading || !app ? (
           <Skeleton variant="text" width={70} />
         ) : (
-          <Tooltip title={app?.name || ''} enterDelay={500}>
-            <Typography noWrap>{app?.name}</Typography>
-          </Tooltip>
+          <AppNameEditable
+            app={app}
+            editing={editingName}
+            setEditing={setEditingName}
+            loading={Boolean(!app)}
+          />
         )}
-        <AppMenu />
+        {app ? (
+          <AppOptions
+            app={app}
+            onRename={handleRename}
+            onDuplicate={() => setDuplicateApp(app)}
+            onDelete={onDelete}
+            dom={dom}
+          />
+        ) : null}
       </Box>
       <Divider />
+      <AppDeleteDialog app={deletedApp} onClose={() => setDeletedApp(null)} redirectOnDelete />
```

Comment:
Would it make sense to include this in the menu as well?

Reply:
Agreed, we can move both the `AppDeleteDialog` and `AppDuplicateDialog` inside the options menu and simplify the `duplicateApp` function as well